src/arch/aarch64/mod.rs

```diff
@@ -133,7 +132,7 @@ pub fn virt_to_phys(
 	// - We are page_aligned, and thus also PageTableEntry aligned.
 	let mut pagetable: &[PageTableEntry] = unsafe {
 		std::mem::transmute::<&[u8], &[PageTableEntry]>(
-			mem.slice_at(pagetable_l0, PAGE_SIZE).unwrap(),
+			mem.slice_at(mem.guest_address, PAGE_SIZE).unwrap(),
```

Comment:
Note for review: The current plan is to remove `mem.guest_address` and read the value from the CR3 register on `x86_64` (on Linux). The fact that it is currently there is a deliberate decision so as to limit the scope of this PR.

Reply:
I have since attempted an implementation with OnceLock (further down below), but could not complete it because I was not sure how I could change the assigned value after the `vm` object is returned.